carma_guidance_plugins/include/carma_guidance_plugins/control_plugin.hpp

```diff
@@ -70,22 +65,29 @@ namespace carma_guidance_plugins
 
 
   public:
+
     /**
-     * \brief ControlPlugin constructor 
+     * \brief ControlPlugin constructor
      */
     explicit ControlPlugin(const rclcpp::NodeOptions &);
 
     //! Virtual destructor for safe deletion
     virtual ~ControlPlugin() = default;
 
+
+    // These callbacks do direct assignment into their respective member variables
+    void current_pose_callback(geometry_msgs::msg::PoseStamped::UniquePtr msg);
+    void current_twist_callback(geometry_msgs::msg::TwistStamped::UniquePtr msg);
+    void current_trajectory_callback(carma_planning_msgs::msg::TrajectoryPlan::UniquePtr msg);
```

Comment:
It looks like these callbacks were made public in order to allow them to be used in unit tests. If this is the case, could we instead define `FRIEND_TEST(...)` within `trajectory_follower_wrapper.hpp` to enable the unit tests to access private members? Reference: https://github.com/usdot-fhwa-stol/carma-platform/blob/11047f2bde3dc2f37b7d2744cd41a3d9772bbc42/pure_pursuit_wrapper/include/pure_pursuit_wrapper/pure_pursuit_wrapper.hpp#L88

Reply:
I think friend tests work when you want to access private members of a class, but they don't allow you to access  the private members of the base class.

Reply:
Oh yes you're right. In that case, rather than changing the private functions to public, could we either:
- Make this unit test a `FRIEND_TEST` of `TrajectoryFollowerWrapperNode` and then directly set `worker_node->current_pose_`, `worker_node->current_twist_`, etc. This is probably preferred so that we don't have to change the base class.
- Or rather than making the `*_callback()` functions public, make them protected in the base class so that this unit test could still access them as a `FRIEND_TEST` of `TrajectoryFollowerWrapperNode`?

Reply:
made them protected, I think these callbacks are important for parts of the plugin and should be accessed for testing. BTW, @adev4a also exposed them as public in his PR, you may want to take a look.